networksharding/kadSharder.go

```diff
@@ -0,0 +1,72 @@
+package networksharding
+
+import (
+	"fmt"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"math/big"
+)
+
+// KadGetShardID Callback returning the shard id of the given peer.ID
+type KadGetShardID func(peer.ID) uint32
+
+// KadSharder KAD based sharder
+//
+// Resets a number of MSb in the distance to nods sharding the same shard
+type KadSharder struct {
```

Comment:
do we need KadSharder exported if the fields are not exported?
we already have a constructor NewKadSharder